goSrc/src/vmCaller/evm/native_contract.go

```diff
@@ -745,6 +783,69 @@ func getRolePermissions(ctx native.Context, args getRolePermissionsArgs) (getRol
 	return getRolePermissionsRets{Result: string(result)}, nil
 }
 
+type getTransactionArgs struct {
+	Hash string
+}
+
+type getTransactionRets struct {
+	Result string
+}
+
+func getTransaction(ctx native.Context, args getTransactionArgs) (getTransactionRets, error) {
+	transaction, err := iroha.GetTransactions(args.Hash)
+	if err != nil {
+		return getTransactionRets{}, err
+	}
+
+	ctx.Logger.Trace.Log("function", "GetAccountTransactions",
+		"hash", args.Hash)
+	result, err := json.Marshal(transaction)
+	return getTransactionRets{Result: string(result)}, nil
+}
+
+type getAccountTxnArgs struct {
+	Account string
+}
+
+type getAccountTxnRets struct {
+	Result string
+}
+
+func getAccountTxn(ctx native.Context, args getAccountTxnArgs) (getAccountTxnRets, error) {
+	transactions, err := iroha.GetAccountTransactions(args.Account)
+	if err != nil {
+		return getAccountTxnRets{}, err
+	}
+
+	ctx.Logger.Trace.Log("function", "GetAccountTransactions",
+		"account", args.Account)
+	result, err := json.Marshal(transactions)
+	return getAccountTxnRets{Result: string(result)}, nil
+}
+
+type getAccountAssetTxnArgs struct {
+	Account string
+	Asset   string
```

Comment:
I'm not golang developer, but maybe You can add here also arguments connected with Your internship project? And also optionally pagination information (page size, order).